### ISSUE TITLE
[5.3] Str::slug and Ampersand

### ISSUE THREAD

**Issue by CupOfTea696** [laravel/framework#14954]
Just a thought, when using `Str::slug` it strips all special characters from a string, but it might be neat for it to replace ampersands with the word `and` (or localised version of the word) rather than just stripping them. Willing to create a PR if you think this is a good idea.

E.g.
```php
// Current behaviour:
$slug = Str::slug('Training & Development'); // training-development

// New behaviour
$slug = Str::slug('Training & Development'); // training-and-development
```

**Comment by GrahamCampbell** [comment]
Could be cool. Would need to go to 5.4 though.

**Comment by morloderex** [comment]
Maybe you should just be converting the string yourself because it presents some security concerns when dealing with speciel characters.

**Comment by GrahamCampbell** [comment]
What concerns?

**Comment by itguy614** [comment]
I think it's a good idea.

**Comment by fernandobandeira** [comment]
Just notice that ``` & ``` translation differs (it's localized) so if this is gonna change we should be able to translate this.

But it would be cool ;)

**Comment by itguy614** [comment]
I'm thinking this might be better handled in the app than in the framework itself.  Then localization could be used.

Something like...

```
$slug = Str::slug(str_replace('&', 'and', 'Training & Development');
```

**Comment by CupOfTea696** [comment]
@fernandobandeira Add a language file to [laravel/laravel](https://github.com/laravel/laravel) and then in the `Str` class check for the presence of a translator, and if it's there try and pull a translation for `and`. When no translator is present or it can't translate, just fall back to english (or maybe don't execute the replacement).

**Comment by CupOfTea696** [comment]
@GrahamCampbell I don't think it necessarily needs to go to 5.4. We could add it but without the translation so that in 5.2 & 5.3 it keeps the default behaviour, unless some one explicitly adds the translation? Just a thought.

**Comment by sisve** [comment]
Localization of single words are very hard to build correctly since there is no sentence context. You will run into agreement (a.k.a concord) which "happens when a word changes form depending on the other words to which it relates" - https://en.wikipedia.org/wiki/Agreement_(linguistics)

That Wikipedia page shows many examples in where it would be impossible to correctly translate single words from another language to English, how would you translate the Swedish "är" (something _is_) into Englishwithout knowing the context? You would be unable to decide between "am" (I am), "are" (you are) or is "He/she/it is".

So, if we cannot build something that can properly support translation to English; what makes us qualified to build something that would support translation to other languages? Even if we limit us to the usual western languages without pictograms and junk (read: Finnish), how would that work out?

From the Wikipedia page; "in Scandinavian languages, adjectives are declined according to the gender, number, and definiteness of the noun they modify. In some cases, predicative adjectives appear to disagree with their subjects. This phenomenon is referred to as Pancake sentences." (I wont go into pancake sentences here, but I found the term funny enough to include)

1. An example with an English source word; "player". What is a player? Are we talking about a video player, hockey player, someone that plays an instrument, a gambler, or someone into dating? They may all be translated differently.

2. There are differences between American English and British English; is it "the team has arrived" or "the team have arrived"?

3. A Swedish-specific issue; "A1 & A2 & B1 & B2 & C1 & C2", where you list pairs of stuff, should become "A1 och A2, B1 och B2 samt C1 och C2" because "samt" is used when dealing with pairs...

Regarding finnish:
http://i.imgur.com/nykjLZY.jpg
http://i.imgur.com/pQREfQw.jpg
https://imgur.com/gallery/NdqH7

TL;DR: Avoid localization of single words.

**Comment by CupOfTea696** [comment]
Third example does show issues with the word "and" so maybe let's not localise this.
